kos_sim/assets.py

```diff
@@ -0,0 +1,88 @@
+"""Asset management for robot models."""
+
+import json
+import os
+import subprocess
+from pathlib import Path
+from typing import Optional
+
+from kos_sim import logger
+
+
+def get_assets_dir() -> Path:
+    """Get the directory containing the assets."""
+    if os.environ.get("KOS_SIM_INSTALLED"):
+        return Path(__file__).parent / "kscale-assets"
+    return Path(__file__).parent.parent / "kscale-assets"
+
+
+def get_available_models() -> list[str]:
+    """Get a list of available model names."""
+    assets_dir = get_assets_dir()
+    if not assets_dir.exists():
+        return []
+    return [d.name for d in assets_dir.iterdir() if d.is_dir()]
+
+
+def ensure_assets_up_to_date() -> bool:
+    """Ensure assets are up to date by checking and updating if necessary."""
+    assets_dir = get_assets_dir()
+    if not assets_dir.exists():
+        logger.error("Assets directory not found at %s", assets_dir)
+        return False
+
+    try:
+        result = subprocess.run(
+            ["git", "submodule", "status"], check=True, capture_output=True, cwd=Path(__file__).parent.parent
```

Comment:
i think the better abstraction is something like `get_kscale_assets_dir()` which returns `Path(__file__).parent.parent` or whatever (i.e., could override it with an environment variable) - will be clearer